File: week1/4.median_of_two_sorted_arrays-两个有序数组中位数/4.median_of_two_sorted_arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function (nums1, nums2) {
    var findMedian = function (nums) {
        let len = nums.length;
        let index = Math.floor(len / 2);
        if (len % 2 == 0) {
            return (nums[index - 1] + nums[index]) / 2.0;
        } else {
            return nums[index];
        }
    }

    let l1 = nums1.length;
    let l2 = nums2.length;
    let lAll = l1 + l2;
    let nums = new Array(lAll);

    if (l1 == 0) {
        return findMedian(nums2);
    }

    if (l2 == 0) {
        return findMedian(nums1);
    }

    let i = 0, j = 0, k = 0;
    while (k != lAll) {
        if (i == l1) {
            while (j != l2) {
                nums[k++] = nums2[j++];
            }
            break;
        }
        if (j == l2) {
            while (i != l1) {
                nums[k++] = nums1[i++];
            }
            break;
        }
        if (nums1[i] > nums2[j]) {
            nums[k++] = nums2[j++];
        } else {
            nums[k++] = nums1[i++];
        }
    }
    return findMedian(nums);
};

File: Algorithms/4.median_of_two_sorted_arrays-两个有序数组中位数/4.median_of_two_sorted_arrays.java
class Solution {
    public double findMedian(int[] nums) {
        int len = nums.length;
        if (len % 2 == 0) {
            return (nums[len / 2 - 1] + nums[len / 2]) / 2.0;
        } else {
            return nums[len / 2];
        }
    }

    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int len1 = nums1.length;
        int len2 = nums2.length;
        int lenAll = len1 + len2;
        int[] nums = new int[lenAll];

        if (len1 == 0) {
            return findMedian(nums2);
        }
        if (len2 == 0) {
            return findMedian(nums1);
        }
        int i = 0;
        int j = 0;
        int k = 0;

        while (k != lenAll) {
            if (i == len1) {
                while (j != len2) {
                    nums[k++] = nums2[j++];
                }
            }
            if (j == len2) {
                while (i != len1) {
                    nums[k++] = nums1[i++];
                }
            }
            if (nums1[i] < nums2[j]) {
                nums[k++] = nums1[i++];
            } else {
                nums[k++] = nums2[j++];
            }
        }
        return findMedian(nums);
    }
}